tests/snuba/search/test_backend.py

```diff
@@ -1217,6 +1229,354 @@ def test_first_release_environments(self):
+
+        # create an issue/group whose events that occur in 2 distinct environments
+
+        group_a_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_a'],
+                'event_id': 'aaa' + ('1' * 29),
+                'message': 'group_a',
+                'environment': 'example_staging',
+                'tags': {
+                    'server': 'example.com',
+                },
+                'timestamp': (release_1_timestamp + timedelta(days=1)).isoformat()[:19],
+                'stacktrace': {
+                    'frames': [{
+                        'module': 'group_a'
+                    }]
+                },
+            },
+            project_id=self.project.id,
+        )
+
+        group_a_event_2 = self.store_event(
+            data={
+                'fingerprint': ['group_a'],
+                'event_id': 'aaa' + ('2' * 29),
+                'message': 'group_a',
+                'environment': 'example_production',
+                'tags': {
+                    'server': 'example.com',
+                },
+                'timestamp': (release_2_timestamp + timedelta(days=1)).isoformat()[:19],
+                'stacktrace': {
+                    'frames': [{
+                        'module': 'group_a'
+                    }]
+                },
+            },
+            project_id=self.project.id,
+        )
+
+        group_a = Group.objects.get(id=group_a_event_1.group.id)
+        assert group_a.id == group_a_event_1.group.id
+        assert group_a.id == group_a_event_2.group.id
+
+        assert group_a.first_seen == group_a_event_1.datetime
+        assert group_a.last_seen == group_a_event_2.datetime
+        assert group_a.last_seen > group_a.first_seen
+
+        group_a.times_seen = 42
+        group_a.status = GroupStatus.UNRESOLVED
+        group_a.save()
+
+        # get the environments for group_a
+
+        prod_env = group_a_event_2.get_environment()
+        staging_env = group_a_event_1.get_environment()
+
+        # create an issue/group whose event that occur in no environments
+        # but will be tied to release_1
+
+        group_b_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_b'],
+                'event_id': 'bbb' + ('1' * 29),
+                'message': 'group_b',
+                'tags': {
+                    'server': 'example.com',
+                },
+                'timestamp': (release_1_timestamp + timedelta(days=1)).isoformat()[:19],
+                'stacktrace': {
+                    'frames': [{
+                        'module': 'group_b'
+                    }]
+                },
+            },
+            project_id=self.project.id,
+        )
+        assert group_b_event_1.get_environment().name == u''  # has no environment
+
+        group_b = Group.objects.get(id=group_b_event_1.group.id)
+        assert group_b.id == group_b_event_1.group.id
```

Comment:
I'm not sure these assertions add much. If this constraint became untrue several other tests would fail.

Reply:
I agree, we don't need to check this. You can also do `group_b = group_b_event_1.group` to get the group a little more easily